core/src/main/scala/scala/pickling/generator/sourcegen.scala

```diff
@@ -111,7 +117,8 @@ private[pickling] trait SourceGenerator extends Macro with tags.FastTypeTagMacro
         case Some(b) =>
           val parentTpe = x.parent.tpe[c.universe.type](c.universe)
           val impl = generatePickleImplFromAst(b)
-          q"""if(classOf[$parentTpe] == picklee.getClass) $impl else $subclasses"""
+          val checkIfParent = q"$classMapper.areSameClasses(picklee.getClass, classOf[$parentTpe])"
+          q"""if($checkIfParent) $impl else $subclasses"""
```

Comment:
Yes, but they're used for different use cases. The one that solves my problem is this one, but for the sake of correctness I decided to put it in the `CaseDef` as well. IIRC, this one takes care of the superclass, the case def of the subclasses.